domain/permission/service/service.go

```diff
@@ -179,16 +186,56 @@ func (s *Service) ReadAllAccessTypeForUser(ctx context.Context, subject string,
 	return userAccess, errors.Trace(err)
 }
 
-// UpdatePermission updates the user's access to the given target to the
-// given access. A NotValid error is returned if the subject (user) string
-// is empty, or the target is not valid. Any errors from the state layer
-// are passed through.
-func (s *Service) UpdatePermission(ctx context.Context, subject string, target permission.ID, access permission.Access) error {
-	if subject == "" {
+type AccessChange string
+
+const (
+	Grant  AccessChange = "grant"
+	Revoke AccessChange = "revoke"
+)
```

Comment:
These feel like they should live in core.

Reply:
agreed